libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -68,17 +68,17 @@ static constexpr uint16_t DELAY_AIRSPEED_TOPIC_MS = AP_DDS_DELAY_AIRSPEED_TOPIC_
 #if AP_DDS_GEOPOSE_PUB_ENABLED
 static constexpr uint16_t DELAY_GEO_POSE_TOPIC_MS = AP_DDS_DELAY_GEO_POSE_TOPIC_MS;
 #endif // AP_DDS_GEOPOSE_PUB_ENABLED
-#if AP_DDS_GOAL_PUB_ENABLED
+#if AP_DDS_GOAL_PUB_ENABLED & AP_SCRIPTING_ENABLED
```

Comment:
If scripting is a requirement for `AP_DDS_GOAL_PUB_ENABLED` then that should have been reflected in setting the defaults for the feature define in `AP_DDS_config.h`.

Reply:
Try https://github.com/ArduPilot/ardupilot/pull/29014